.concourse/tasks/upgrade.sh

```diff
@@ -15,10 +15,40 @@ export FORCE_DELETE=true
 export HELM_VERSION="v3.1.1"
 export BACKEND=imported
 export QUIET_OUTPUT=true
-CLUSTER_NAME="$(cat kind-environments/name)"
-export CLUSTER_NAME
-KUBECFG="$(readlink -f kind-environments/metadata)"
-export KUBECFG
+export GKE_CLUSTER_NAME="kubecf-ci-$(cat semver.gke-cluster/version | sed 's/\./-/g')"
+export KUBECFG="$(readlink -f ~/.kube/config)"
+
+printf "%s" '((gke-suse-cap-json))' > $PWD/gke-key.json
+export GKE_CRED_JSON=$PWD/gke-key.json
+gcloud auth activate-service-account --key-file $PWD/gke-key.json
+
+export GKE_PROJECT={{ if has . "gke_project" }}{{ .gke_project }}{{ else }}"suse-225215"{{ end }}
+export GKE_ZONE={{ if has . "gke_zone" }}{{ .gke_zone }}{{ else }}"europe-west3-c"{{ end }}
+export GKE_CLUSTER_NAME="kubecf-ci-$(cat semver.gke-cluster/version)"
+
+gcloud --quiet beta container \
+  --project "${GKE_PROJECT}" clusters create "${GKE_CLUSTER_NAME}" \
+  --zone "${GKE_ZONE}" \
+  --no-enable-basic-auth \
+  --machine-type "n1-highcpu-16" \
+  --image-type "UBUNTU" \
+  --disk-type "pd-ssd" \
+  --disk-size "100" \
+  --metadata disable-legacy-endpoints=true \
+  --scopes "https://www.googleapis.com/auth/devstorage.read_only","https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/monitoring","https://www.googleapis.com/auth/servicecontrol","https://www.googleapis.com/auth/service.management.readonly","https://www.googleapis.com/auth/trace.append" \
+  --preemptible \
+  --num-nodes "1" \
+  --enable-stackdriver-kubernetes \
+  --enable-ip-alias \
+  --network "projects/${GKE_PROJECT}/global/networks/default" \
+  --subnetwork "projects/${GKE_PROJECT}/regions/$(echo ${GKE_ZONE} | sed 's/-.$//')/subnetworks/default" \
+  --default-max-pods-per-node "110" \
+  --no-enable-master-authorized-networks \
+  --addons HorizontalPodAutoscaling,HttpLoadBalancing \
+  --no-enable-autorepair
+
+# Get a kubeconfig
+gcloud container clusters get-credentials "${GKE_CLUSTER_NAME}" --zone "${GKE_CLUSTER_ZONE}" --project "${GKE_PROJECT}"
```

Comment:
Catapult is doing some strange things here: https://github.com/SUSE/catapult/blob/master/backend/gke/kubeconfig.sh#L6

it requires a kubeconfig but then [gets one itself](https://github.com/SUSE/catapult/blob/master/backend/gke/kubeconfig.sh#L35). We fetch the kubeconfig here too to make catapult happy but this needs some clarification on how catapult is supposed to work with gke.

Reply:
This seems to come from https://github.com/SUSE/catapult/pull/183 , which is a fix as it seems that `gcloud` creates configs with expiry, but I actually think that doesn't hold true anymore. @prabalsharma , are we sure about that bit?